src/ServiceFabric.Mocks/MockReliableStateManager.cs

```diff
@@ -72,6 +75,36 @@ public Task TriggerRestoreCompleted()
         }
         #endregion
 
+        /// <summary>
```

Comment:
I wasn't a huge fan of these test hooks. I was worried that if someone wanted to do a long running simulation with the mocks that the dictionary would grow very large storing unneeded information. I'm also not sure why back-compat for test hooks is necessary. 

I think if this detail is wanted in the hooks, I'd rather change the MockTransactionChanged event to take an enum {Abort,Create,Commit} instead of the bool isCommit. Then write a TransactionWatcher handler that you can hook up to the (new) MockTransactionChanged event to implement the functionality of these hooks when needed.

Reply:
I can follow-up with a PR with these changes if you'd like.